java/src/main/java/com/google/appengine/tools/mapreduce/impl/shardedjob/ShardedJobRunner.java

```diff
@@ -119,87 +107,81 @@ public static RetryerBuilder getRetryerBuilderAggressive() {
 
 
 
-  private ShardedJobStateImpl<T> lookupJobState(Transaction tx, String jobId) {
-    try {
-      Entity entity = DATASTORE.get(tx, ShardedJobStateImpl.ShardedJobSerializer.makeKey(jobId));
-      return ShardedJobStateImpl.ShardedJobSerializer.fromEntity(entity);
-    } catch (EntityNotFoundException e) {
-      return null;
-    }
+  private ShardedJobStateImpl<T> lookupJobState(@NonNull Transaction tx, String jobId) {
+    return (ShardedJobStateImpl<T>) Optional.ofNullable(tx.get(ShardedJobStateImpl.ShardedJobSerializer.makeKey(tx.getDatastore(), jobId)))
+      .map(in -> ShardedJobStateImpl.ShardedJobSerializer.fromEntity(tx, in))
+      .orElse(null);
   }
 
   @VisibleForTesting
-  IncrementalTaskState<T> lookupTaskState(Transaction tx, String taskId) {
-    try {
-      Entity entity = DATASTORE.get(tx, IncrementalTaskState.Serializer.makeKey(taskId));
-      return IncrementalTaskState.Serializer.fromEntity(entity);
-    } catch (EntityNotFoundException e) {
-      return null;
+  IncrementalTaskState<T> lookupTaskState(@NonNull Transaction tx, String taskId) {
+    return (IncrementalTaskState<T>) Optional.ofNullable(tx.get(IncrementalTaskState.Serializer.makeKey(tx.getDatastore(), taskId)))
```

Comment:
can the `EntityNotFoundException` still happen or not? If yes, how will this behave?